htdocs/societe/class/societe.class.php

```diff
@@ -1479,7 +1479,7 @@ public function update($id, $user = '', $call_trigger = 1, $allowmodcodeclient =
 
 			$sql .= ",fk_effectif = ".($this->effectif_id > 0 ? ((int) $this->effectif_id) : "null");
 			if (isset($this->stcomm_id)) {
-				$sql .= ",fk_stcomm=".($this->stcomm_id > 0 ? ((int) $this->stcomm_id) : "0");
+				$sql .= ",fk_stcomm=".(($this->stcomm_id > 0 || $this->stcomm_id = -1) ? ((int) $this->stcomm_id) : "0");
```

Comment:
You mean == and not = ?
May be you replace with just    ...fk_stcomm=".((int) $this->stcomm_id);    ?

Reply:
@eldy yes you are right, by dint of fixing bugs we get confused!
it's fixed